web-client/public/js/api/grnsight-api.js

```diff
@@ -19,6 +19,36 @@ const buildExpressionURL = function (selection, genes) {
     baseQuery;
 };
 
+
+const buildNetworkGenesQuery = (genes) => {
+    let result = "";
+    if (Array.isArray(genes)) {
+        for (let gene of genes) {
+            result += `${gene},`;
+        }
+    } else {
+        for (let gene in genes) {
+            result += `${gene},`;
+        }
+    }
```

Comment:
Isn’t the first case doable with `.join`? Also, what data structure does the `else` case cover—an object? An iterable?

Looking to see how we can consolidate this

Reply:
Sometimes the genes are passed in as an object with keys, and other times the genes are passed in as an array, so if it's an object you can iterate through the keys with `let gene in genes`, and if it's an array you can iterate through the genes as is with `let gene of genes`.